doc/lag/Increasing LACP PDU timeout during warm-reboot.md

```diff
@@ -0,0 +1,119 @@
+
+# Protocol
+
+To change the number of retries, an Ethernet packet of the fillowing structure
+will be sent. This Ethernet packet will have an ethertype of 0x6300, and will
+not have an IPv4 or IPv6 layer on top of it. Instead, there will instead be
+multiple TLV fields, similar to LACP.
+
+The TLV types will be defined as follows:
+
+| Value | Description         |
+|-------|---------------------|
+| 0x01  | Actor Information   |
+| 0x02  | Partner Information |
+| 0x03  | Retry Count         |
+
+Both Actor Information and Partner Information have the following content:
+
+| Starting byte | Length | Description     |
+|---------------|--------|-----------------|
+|      0        |   2    | System Priority |
+|      2        |   6    | System ID       |
+|      8        |   2    | Key             |
+|     10        |   2    | Port Priority   |
+|     12        |   2    | Port            |
+|     14        |   2    | Padding         |
+
+Retry count have the following content:
+
+| Starting byte | Length | Description     |
+|---------------|--------|-----------------|
+|      0        |   2    | New retry count |
+|      2        |   2    | Padding         |
+
+When the retry count needs to be changed, the sending device must send a packet
+with ethertype 0x6300, and the data will contain the Actor Information, Partner
+Information, and Retry Count TLVs. The receiving device must validate the actor
+and partner information, and then update the retry count as specified. No
+acknowledgment packet is sent back.
+
+The custom ethertype is so that if the packet is sent to non-SONiC devices
+or SONiC devices running an older version, then it can be silently discarded,
+without it getting incorrectly handled by the peer device and without it getting
+forwarded to a different device.
+
+# Changing Max Retries for Warmboot
+
+As part of a SONiC device starting the warmboot process, LACP PDUs are sent to
+all of the peers, to refresh the timers on the peers. This allows the warmboot
+process the full 90 seconds for control plane to come back up and for PDUs to be
+sent again after warmboot.
+
+Now, in addition to refreshing the PDUs timer, the above-specified Ethernet
+packet (with ethertype 0x6300) will be sent to the peer devices, with the new
+retry count set to 5. This notifies the peer device that for this device that
+for this LAG, it should allow up to 5 retries before bringing down this LAG. If
+the peer device is not running SONiC or is not running a version of SONiC, then
+the expectation is that the packet will just be dropped at the peer device with
+no handling. In either case, teamd will not wait for any acknowledgment packet.
+
+After warmboot is done, and teamd has started up after warmboot, it will send
+the above-specified Ethernet packet again, but this time, the new retry count is
+set to 3, thus restoring it to the standard value.
+
+# CLI
+
+No new CLI options or config options will be added, as this is not meant to be
```

Comment:
Config knob may be required to avoid sending unnecessary 0x6300 packets during warm-reboot when SONiC is connected to Non-SONiC device.

Reply:
I would like to see a configurable option to enable this feature, default should be left disabled with no custom TLV to interfere with WB. The HLD diverged from standard LACP protocol definition, using custom TLV to overcome SONiC WB timing,  and potentially a phase2 send and ack mechanism in the future that could potentially block WB from proceeding. For deployment where non standard protocol packets are forbidden, we need a configurable option to control this behavior. Preferably leaving it disable by default.

Reply:
The HLD has been updated with a new design/approach. By default, this feature is disabled, so there won't be any custom packets unless configured.